bot/exts/info/information.py

```diff
@@ -533,21 +553,43 @@ async def rules(self, ctx: Context, rules: Greedy[int]) -> None:
             await ctx.send(embed=rules_embed)
             return
 
-        full_rules = await self.bot.api_client.get("rules", params={"link_format": "md"})
-
         # Remove duplicates and sort the rule indices
-        rules = sorted(set(rules))
+        rule_numbers = sorted(set(rule_numbers))
+        # Remove duplicate keywords and preserve the order of initial keywords
+        keywords = list(dict.fromkeys(keywords))
 
-        invalid = ", ".join(str(index) for index in rules if index < 1 or index > len(full_rules))
+        invalid = ", ".join(
+            str(rule_number) for rule_number in rule_numbers
+            if rule_number < 1 or rule_number > len(full_rules))
 
         if invalid:
             await ctx.send(shorten(":x: Invalid rule indices: " + invalid, 75, placeholder=" ..."))
             return
 
-        for rule in rules:
-            self.bot.stats.incr(f"rule_uses.{rule}")
+        final_rule_numbers = []
+        final_rule_numbers.extend(rule_numbers)
+
+        for keyword in keywords:
+            for pick, rule in enumerate(full_rules):
+                if keyword in rule[1]:
+                    final_rule_numbers.append(pick + 1)
+                    break
```

Comment:
I think it's slightly neater here to use the `start` argument of enumerate, and I think `rule_number` is a more descriptive name than `pick`.
```suggestion
        for keyword in keywords:
            for rule_number, rule in enumerate(full_rules, 1):
                if keyword in rule[1]:
                    final_rule_numbers.append(rule_number)
                    break
```

Where you create `available_keywords` above you could instead consider making a `keyword_to_rule_number` dictionary, which you could still use `in` on as you do for `available_keywords`, but would also allow you simplify this a little.
```python
for keyword in keywords:
    if rule_number := keyword_to_rule_number.get(keyword):
        final_rule_numbers.append(rule_number)
```
Not a requirement though